operators/src/util/raster_stream_to_geotiff.rs

```diff
@@ -150,15 +155,15 @@ struct GdalDatasetWriter<P: Pixel + GdalType> {
     gdal_tiff_options: GdalGeoTiffOptions,
     gdal_tiff_metadata: GdalGeoTiffDatasetMetadata,
     output_bounds: SpatialPartition2D,
-    output_geo_transform: GeoTransform,
-    x_pixel_size: f64,
-    y_pixel_size: f64,
     _type: std::marker::PhantomData<P>,
     use_big_tiff: bool,
+    global_offset: GridIdx2D,
+    global_limit: GridIdx2D,
 }
 
 impl<P: Pixel + GdalType> GdalDatasetWriter<P> {
     fn new(
+        tiling_specification: TilingSpecification,
```

Comment:
to keep the write more easy to reuse please add the offset and limit as prams to the `new` method and add a `new_with_tiling_spec` or something where they are calculated (and then call `new`).